spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/MessageHistoryTrackerConfigurer.java

```diff
@@ -40,17 +42,31 @@
 
 	private final ChannelBindingServiceProperties channelBindingServiceProperties;
 
-	@Autowired
-	MessageBuilderFactory messageBuilderFactory;
+	private final MessageBuilderFactory messageBuilderFactory;
 
-	public MessageHistoryTrackerConfigurer(ChannelBindingServiceProperties channelBindingServiceProperties) {
+	public MessageHistoryTrackerConfigurer(ChannelBindingServiceProperties channelBindingServiceProperties,
+			MessageBuilderFactory messageBuilderFactory) {
 		this.channelBindingServiceProperties = channelBindingServiceProperties;
+		this.messageBuilderFactory = messageBuilderFactory;
 	}
 
 	@Override
 	public void configureMessageChannel(MessageChannel messageChannel, String channelName) {
 		BindingProperties bindingProperties = channelBindingServiceProperties.getBindings().get(channelName);
 		if (bindingProperties != null && Boolean.TRUE.equals(bindingProperties.isTrackHistory())) {
+			final Set<String> trackHistoryProperties = StringUtils.commaDelimitedListToSet(bindingProperties.getTrackHistoryProperties());
```

Comment:
We should probably check for an empty array of tracked properties too. What should be the behaviour in that case?